sdk/scheduler/src/main/java/com/mesosphere/sdk/state/StateStoreUtils.java

```diff
@@ -256,6 +290,33 @@ private static void setBooleanProperty(StateStore stateStore, String propertyNam
     }
 
     /**
+     * Stores a TaskStatus as a Property in the provided state store.
+     */
+    public static void storeTaskStatusAsProperty(StateStore stateStore, String taskName, TaskStatus taskInfo)
+            throws StateStoreException {
+        stateStore.storeProperty(taskName + PROPERTY_TASK_INFO_SUFFIX, taskInfo.toByteArray());
+    }
+
+    /**
+     * Returns an Optional<TaskStatus> from the properties in the provided state store for the specified
+     * task name.
+     */
+    public static Optional<TaskStatus> getTaskStatusFromProperty(StateStore stateStore, String taskName)
+            throws StateStoreException {
```

Comment:
It could throw on the stateStore.fetchPropertyKeys. But, I'm gonna refactor into what you suggested.